pkg/util/provider/machinecodes/status/status.go

```diff
@@ -88,10 +107,18 @@ func FromError(err error) (s *Status, ok bool) {
 
 	if matches, errInFind := findInString(err.Error()); errInFind == nil {
 		code := codes.StringToCode(matches[0])
-		return New(code, matches[1]), true
+		return &Status{
+			code:    int32(code),
+			message: matches[1],
+			cause:   err,
+		}, true
```

Comment:
```suggestion
		return WrapError(code, matches[1], err), true
```

Something like this should be equivalent,

Reply:
It is golang good practices to keep the keys. This will also be highlighted and often auto-corrected by goland IDE as well. Let us keep it this way as it also helps with readability.

Reply:
Fair enough, lets keep it that way then
Thanks!